packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx

```diff
@@ -120,144 +177,73 @@ const ClinicalResultBloodPressure = (props) => {
   let clinicalResultBloodPressureDisplay = null;
 
   if (hasResultError) {
-    clinicalResultBloodPressureDisplay = <ResultError />;
-  } else if (hasResultNoData) {
-    clinicalResultBloodPressureDisplay = <NoData />;
-  } else {
-    const compareUnits = {};
-    const compareStatusIsInError = {};
-
-    const compareConceptDisplays = {
-      originalSystolic: systolic && systolic.conceptDisplay,
-      originalDiastolic: diastolic && diastolic.conceptDisplay,
-    };
-    const compareDatetimeDisplays = {
-      originalSystolic: systolic && systolic.datetimeDisplay,
-      originalDiastolic: diastolic && diastolic.datetimeDisplay,
-    };
+    return <ResultError />;
+  }
+  if (hasResultNoData) {
+    return <NoData />;
+  }
 
-    let iconGroupDisplayElement = null;
+  const systolicResult = createResult(systolic);
+  const diastolicResult = createResult(diastolic);
 
-    const noDataSystolic = (systolic && systolic.resultNoData === true);
-    if (systolic && !noDataSystolic) {
-      const {
-        result,
-        status,
-        conceptDisplay,
-        datetimeDisplay,
-      } = systolic;
+  const conceptDisplayElement = createConceptDisplays({
+    originalSystolic: systolicResult.conceptDisplay,
+    originalDiastolic: diastolicResult.conceptDisplay,
+    systolic: systolicResult.cleanedConceptDisplay,
+    diastolic: diastolicResult.cleanedConceptDisplay,
+  });
 
-      if (!isEmpty(result.unit)) { compareUnits.systolic = result.unit.trim().toLowerCase(); }
-      if (!isEmpty(status)) { compareStatusIsInError.systolic = checkIsStatusInError(status); }
-      if (!isEmpty(conceptDisplay)) { compareConceptDisplays.systolic = conceptDisplay.trim().toLowerCase(); }
-      if (!isEmpty(datetimeDisplay)) { compareDatetimeDisplays.systolic = datetimeDisplay.trim().toLowerCase(); }
-    }
+  const datetimeDisplayElement = createDatetimeDisplays({
+    originalSystolic: systolicResult.datetimeDisplay,
+    originalDiastolic: diastolicResult.datetimeDisplay,
```

Comment:
do we need to check against the original values? I would have expected the cleaned values would be equivalent... It might even be better for this check: https://github.com/cerner/terra-clinical/blob/86ab6dbbc5f074c20783c6f4830136bb7a43f63c/packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx#L93
Incase there's trailing whitespace

Reply:
actually nvm, in the method I linked, systolic and diastolic do not map to the original objects